### PR TITLE
Update basic_maths.py

### PR DESCRIPTION
**Pull request by nivid26**
### Describe your change:

* [ ] Add an algorithm?
* [X] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
Please add a test that fails under the current code but passes under the new code so we have proof that this change is really needed.

**Comment by nivid26** [comment]
Old Code:
number_of_divisors(6) return 2 but actually number of divisor of 6 is 4(1,2,3,6).
New Code:
number_of_divisors(6) return 4 which right answer.

File: maths/basic_maths.py [hunk 1]
```diff
@@ -57,6 +57,8 @@ def number_of_divisors(n: int) -> int:
             temp += 1
             n = int(n / i)
         div *= temp
+    if n > 1:
+        div *= 2
```
Comment (poyea) [inline maths/basic_maths.py:61]: Could you come up with an example to show why this is needed? 🤔

**Comment by poyea** [comment]
@nivid26 This actually looks correct. Would you be able to add a few tests to cover this?

**Comment by nivid26** [comment]
Number  of Divisor
------------------------------------------------------------------------------------
Number | Old   code return values | New   Code return value | Correct   value
     3       |                1                       |                2                       |          2
    6        |                2                       |               4                        |          4
    7        |                1                       |               2                        |          2
  10        |                2                       |               4                        |          4
  11        |                1                       |               2                        |          2
  12        |                3                       |               6                        |          6

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

**Pull request merged**